lib/portlayer/storage/image_cache.go

```diff
@@ -300,3 +298,54 @@ func (c *NameLookupCache) ListImages(ctx context.Context, store *url.URL, IDs []
 
 	return imageList, nil
 }
+
+// DeleteImage deletes an image from the image store.  If it is in use or is being inheritted from, then this will return an error.
+func (c *NameLookupCache) DeleteImage(ctx context.Context, image *Image) error {
+	infof("DeleteImage: deleting %s", image.Self())
+
+	// Check the image exists.  This will rehydrate the cache if necessary.
+	_, err := c.GetImage(ctx, image.Store, image.ID)
+	if err != nil {
+		return errorf("DeleteImage: %s", err)
+	}
+
+	// get the relevant cache
+	c.storeCacheLock.Lock()
+	indx := c.storeCache[*image.Store]
+	c.storeCacheLock.Unlock()
+
+	hasChildren, err := indx.HasChildren(image.Self())
+	if err != nil {
+		return errorf("DeleteImage: %s", err)
+	}
+
+	if hasChildren {
+		return errorf("deleting image %s: in use by child images", image.Self())
+	}
+
+	// The datastore will tell us if the image is attached
```

Comment:
Only true if the VM with it attached is powered on, unless there's logic in the datastore call?